src/analysis/convex/between.lean

```diff
@@ -0,0 +1,500 @@
+/-
+Copyright (c) 2022 Joseph Myers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joseph Myers
+-/
+import analysis.convex.segment
+import linear_algebra.affine_space.finite_dimensional
+
+/-!
+# Betweenness in affine spaces
+
+This file defines notions of a point in an affine space being between two given points.
+
+## Main definitions
+
+* `affine_segment R x y`: The segment of points weakly between `x` and `y`.
+* `wbtw R x y z`: The point `y` is weakly between `x` and `z`.
+* `sbtw R x y z`: The point `y` is strictly between `x` and `z`.
+
+-/
+
+variables (R : Type*) {V V' P P' : Type*}
+
+open affine_map
+
+section ordered_ring
+
+variables [ordered_ring R] [add_comm_group V] [module R V] [add_torsor V P]
+variables [add_comm_group V'] [module R V'] [add_torsor V' P']
+
+include V
+
+/-- The segment of points weakly between `x` and `y`. When convexity is refactored to support
+abstract affine combination spaces, this will no longer need to be a separate definition from
+`segment`. However, lemmas involving `+ᵥ` or `-ᵥ` will still be relevant after such a
+refactoring, as distinct from versions involving `+` or `-` in a module. -/
+def affine_segment (x y : P) := line_map x y '' (set.Icc (0 : R) 1)
+
+lemma affine_segment_eq_segment (x y : V) : affine_segment R x y = segment R x y :=
+by rw [segment_eq_image_line_map, affine_segment]
+
+lemma affine_segment_symm (x y : P) : affine_segment R x y = affine_segment R y x :=
```

Comment:
Sorry for the late reply. Yes, please change this to `_comm`.
```suggestion
lemma affine_segment_comm (x y : P) : affine_segment R x y = affine_segment R y x :=
```

Reply:
I've done that renaming as well.